sql/core/src/test/scala/org/apache/spark/sql/QueryTest.scala

```diff
@@ -168,6 +191,28 @@ object QueryTest {
     return None
   }
 
+  /**
+   * Runs the plan and makes sure the answer is within absTol of the expected result.
+   * @param actualAnswer the actual result in a [[Row]].
+   * @param expectedAnswer the expected result in a[[Row]].
+   * @param absTol the absolute tolerance between actual and expected answers.
+   */
+  protected def checkAggregatesWithTol(actualAnswer: Row, expectedAnswer: Row, absTol: Double) = {
```

Comment:
@holdenk Changed name to `checkAggregatesWithTol` and moved to query test. We can now use this for the various aggregate testing. I think ultimately a more elegant solution is needed, but this provides a way to add tolerancing for now.